mantis-control-plane/mantis-control-plane-core/src/main/java/io/mantisrx/server/core/master/ZookeeperMasterMonitor.java

```diff
@@ -36,36 +37,45 @@
 /**
  * A monitor that monitors the status of Mantis masters.
  */
-public class ZookeeperMasterMonitor extends AbstractIdleService implements MasterMonitor {
+public class ZookeeperMasterMonitor extends BaseService implements MasterMonitor {
```

Comment:
Moving to `BaseService` makes this feel more consistent, but if the maintainers were moving away from `BaseService` to `AbstractIdleService` then I need to reverse my choices here. In addition, this is where I felt the urge to change the monitor factory interface and ultimately the `MasterMonitor` interface to become an abstract class of `BaseService`. At the moment I'm satisfied with the choice to stop at an interface versus an abstract class.